pkg/commands/deploy.go

```diff
@@ -17,35 +17,107 @@
 package commands
 
 import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/fromanirh/deployer/pkg/clientutil"
+	"github.com/fromanirh/deployer/pkg/manifests"
 	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	"io/fs"
+	"io/ioutil"
 )
 
-type deployOptions struct{}
+type deployOptions struct{
+	crdPath string
+}
 
 func NewDeployCommand(commonOpts *CommonOptions) *cobra.Command {
-	//	opts := &deployOptions{}
+	opts := &deployOptions{}
 	deploy := &cobra.Command{
 		Use:   "deploy",
 		Short: "deploy the components and configurations needed for topology-aware-scheduling",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			// TODO: deploy components here
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+
+			if err := opts.Deploy(); err != nil {
+				return err
+			}
 			return nil
 		},
 		Args: cobra.NoArgs,
 	}
+	opts.AddFlags(deploy.Flags())
 	return deploy
 }
 
 func NewRemoveCommand(commonOpts *CommonOptions) *cobra.Command {
-	//	opts := &deployOptions{}
+	opts := &deployOptions{}
 	remove := &cobra.Command{
 		Use:   "remove",
 		Short: "remove the components and configurations needed for topology-aware-scheduling",
 		RunE: func(cmd *cobra.Command, args []string) error {
 			// TODO: remove components here
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+
+			if err := opts.Remove(); err != nil {
+				return err
+			}
 			return nil
 		},
 		Args: cobra.NoArgs,
 	}
+	opts.AddFlags(remove.Flags())
```

Comment:
what happens if we don't `AddFlags` explicitely?

Reply:
like, what if we do like https://github.com/fromanirh/deployer/blob/main/pkg/commands/validate.go#L44

Reply:
I don't know if it's possible let me check